pkg/client-sdk/covenantless_client.go

```diff
@@ -1047,34 +1050,32 @@ func (a *covenantlessArkClient) Settle(ctx context.Context) (string, error) {
 func (a *covenantlessArkClient) GetTransactionHistory(
 	ctx context.Context,
 ) ([]types.Transaction, error) {
-	if a.Config.WithTransactionFeed {
-		return a.store.TransactionStore().GetAllTransactions(ctx)
-	}
-
 	if a.Config == nil {
 		return nil, fmt.Errorf("client not initialized")
 	}
 
+	if a.Config.WithTransactionFeed {
+		return a.store.TransactionStore().GetAllTransactions(ctx)
+	}
+
 	spendableVtxos, spentVtxos, err := a.ListVtxos(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	boardingTxs, ignoreVtxos, err := a.getBoardingTxs(ctx)
+	boardingTxs, _, err := a.getBoardingTxs(ctx)
```

Comment:
```suggestion
	boardingTxs, err := a.getBoardingTxs(ctx)
```

the second return var is never used, let's remove it